src/frontends/android/apiclient/src/main/java/org/strongswan/android/apiclient/ReturnMessenger.java

```diff
@@ -0,0 +1,64 @@
+/*
+ * Copyright © 2015 FancyFon Software Ltd.
+ * All rights reserved.
+ */
+package org.strongswan.android.apiclient;
+
+import android.content.Context;
+import android.os.Bundle;
+import android.os.Handler;
+import android.os.Message;
+import android.os.Messenger;
+import com.google.inject.Inject;
+
+/**
+ * @author Marcin Waligórski <marcin.waligorski@fancyfon.com>
+ */
+public class ReturnMessenger {
+    private static final String TAG = ReturnMessenger.class.getSimpleName();
+
+    @Inject
+    Context context;
+    @Inject
+    Logger logger;
+
+    public Messenger getReturnMessenger() {
+        return returnMessenger;
+    }
+
+    private Messenger returnMessenger = new Messenger(new Handler() {
+        @Override
+        public void handleMessage(Message msg) {
```

Comment:
for this whole method I prefer if-return style that if-else,  For me its more deadable, but it is a matter of style only